src/lib/db/client-applications-store.ts

```diff
@@ -35,8 +36,49 @@ const mapRow: (any) => IClientApplication = (row) => ({
     icon: row.icon,
     lastSeen: row.last_seen,
     announced: row.announced,
+    project: row.project,
+    environment: row.environment,
 });
 
+const reduceRows = (rows: any[]): IClientApplication[] => {
+    const appsObj = rows.reduce((acc, row) => {
+        const existingApp = acc[row.app_name];
+
+        if (existingApp) {
+            let project = existingApp.usage.find(
+                (usage) => usage.project === row.project,
+            );
+
+            if (project) {
+                project.environments.push(row.environment);
+            } else {
+                existingApp.usage.push({
+                    project: row.project,
+                    environments: [row.environment],
+                });
+            }
+        } else {
+            const mappedRow = mapRow(row);
+            acc[row.app_name] = {
+                ...mappedRow,
+                usage:
+                    row.project && row.environment
+                        ? [
+                              {
+                                  project: row.project,
+                                  environments: [row.environment],
+                              },
+                          ]
+                        : [],
+            };
```

Comment:
Spent a fair bit of time looking at this function to figure out what it does. Could we simplify it?

Reply:
Refactored a little, but cant do much more.